### DISCUSSION TITLE
useNavigation or props

### DISCUSSION THREAD

**Discussion by shamilovtim** [react-navigation/react-navigation#9550] [Q&A]
When props are available should useNavigation be avoided? Which should be preferred when both are available? Are there measurable performance differences from using the hook vs. the prop?

**Comment by raajnadar** [comment]
Whenever possible use hooks this is what I would recommend. Because easy to use & more flexible than props.

**Reply by shamilovtim** [reply]
Thanks! Is there no performance drawback from using the hook vs. getting the reference straight off the prop?

**Reply by raajnadar** [reply]
Oh sorry I missed your comment! The advantage of `useNavigation` hooks is it always stays at the top level. And props coming from a component will be at component level.

Eg- The `goBack()` from `useNavigation` and from props might be different in some cases when using nested navs.

**Comment by flexbox** [comment]
@shamilovtim That's one of my existential questions as well 😅.

I am in the process of adding a LOT of tests in my current app (unit testing with [`react-native-testing-library`](https://github.com/callstack/react-native-testing-library) and UI with [`storybook`](https://github.com/storybookjs/react-native)).
I noticed it's easier to work with the hook, as @raajnadar said.
